lib/oli_web/live/delivery/student/content_live.ex

```diff
@@ -3,23 +3,391 @@ defmodule OliWeb.Delivery.Student.ContentLive do
 
   import OliWeb.Components.Delivery.Layouts
 
+  alias OliWeb.Common.FormatDateTime
+  alias Phoenix.LiveView.JS
+  alias OliWeb.Components.Modal
+
   def mount(_params, _session, socket) do
-    {:ok, socket}
+    # TODO replace this with a call to the Cache data in ETS
+    # (finally it will be stored as a json in a new section field maybe called "cached_hierarchy")
+    hierarchy =
+      Oli.Publishing.DeliveryResolver.full_hierarchy(socket.assigns.section.slug)
+
+    {:ok,
+     assign(socket,
+       hierarchy: hierarchy,
+       selected_unit_uuid: nil,
+       selected_module: nil,
+       selected_module_index: nil
+     )}
+  end
+
+  def handle_event(
+        "select_module",
+        %{
+          "unit_uuid" => unit_uuid,
+          "module_uuid" => module_uuid,
+          "module_index" => selected_module_index
+        },
+        socket
+      ) do
+    socket =
+      if module_uuid == (socket.assigns.selected_module && socket.assigns.selected_module.uuid) do
```

Comment:
Yes, that was my first approach but, since the selected_module is a HierarchyNode struct that does not implement the Access behaviour, I ended up with the current approach.
But, I just remembered we can use [Access.key!](https://hexdocs.pm/elixir/1.15.5/Access.html#key!/1) so I ended up with
`if module_uuid == get_in(socket.assigns, [:selected_module, Access.key!(:uuid)]) do`